lib/bot.js

```diff
@@ -137,11 +137,17 @@ class Bot {
     logger.debug('Channel Mapping', channelName, this.channelMapping[channelName]);
     if (ircChannel) {
       const username = author.username;
+      let nickname = author.username;
+      const fromServer = message.server;
+      if (fromServer != null) {
+        const userDetails = fromServer.detailsOfUser(author);
+        if (userDetails != null && userDetails.nick != null) nickname = userDetails.nick;
```

Comment:
could do `nickname = _.get(userDetails, 'nick', nickname)` if you want. See https://lodash.com/docs#get